File: java/src/main/java/com/github/knifeofdreams/codekata/CodeKata.java
package com.github.knifeofdreams.codekata;

import java.util.function.IntSupplier;

public class CodeKata {

  private final IntSupplier randomNumberSupplier;

  public CodeKata(IntSupplier randomNumberSupplier) {
    this.randomNumberSupplier = randomNumberSupplier;
  }

  public int generateRandomNumber() {
    return randomNumberSupplier.getAsInt();
  }

}

File: kotlin/src/main/kotlin/com/github/knifeofdreams/codekata/kotlin/CodeKata.kt
package com.github.knifeofdreams.codekata.kotlin

class CodeKata(val intSupplier: () -> Int) {

    fun generateRandomNumber(): Int {
        return intSupplier.invoke()
    }
}

fun generateRandomNumber(supplier: () -> Int): Int {
    return supplier.invoke()
}
